Projects/My_website/myWebsite/blog/views.py

```diff
@@ -26,16 +26,58 @@ class userBlog(APIView):
     authentication_classes = [TokenAuthentication]
     permission_classes = [IsAuthenticated]
 
+    def get_item(self,slug):
+        item = Post.objects.get(slug=slug)
+        if item:
+            return item
+        else:
+            return False
+        
     def get(self,request):
         posts = Post.objects.filter(author=request.user.id)
         serializers = blogSerializer(posts,many=True)
         return Response(serializers.data)
     
-    def put(self,request):
-        return Response({'message':"this is put method (under devlopment)"})
-    def delete(self,request):
-        return Response({'message':"this is get method (under devlopment)"})   
-    def patch(self,request):
-        return ({'message':"this is patch method (under devlopment)"})
+    def post(self,request):
+        data = request.data
+        data['author']=request.user.id
+        serializers = blogSerializer(data=data)
+        if serializers.is_valid():
+            serializers.save()
+            return Response(serializers.data)
+        return Response(serializers.errors)
+    
+    def put(self,request,slug):
+        data = request.data
+        if data['author'] != request.user.id:
+            return Response({'message':"you are not valid author"})
+        obj = self.get_item(slug)
+        if obj==False : return Response({'message':"in valid slug"})
+        serializer = blogSerializer(obj,data=data)
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data)
+        return Response(serializer.errors)
+    
+    def patch(self,request,slug):   
+        data = request.data
+        if data['author'] != request.user.id:
+            return Response({'message':"you are not valid author"})
```

Comment:
SUGGESTION : Pls use more generic messages here. Like `You are not authorised to perform this action.`